addons/project/static/src/views/project_task_form/project_task_form_controller.js

```diff
@@ -47,7 +47,7 @@ export class ProjectTaskFormController extends FormController {
         this.dialogService.add(
             HistoryDialog,
             {
-                recordId: this.props.resId,
+                recordId: this.props.resId || this.model.config.resId,
```

Comment:
hello, thanks for your work, but actually, you're not ensuring that resId has value like you wrote in the description, what happens if this.model.config.resId is null (for any reason), it will crash again, so maybe we can add a condition

```suggestion
               const recordId = this.props.resId || this.model.config.resId;
               if (!recordId) {
                   return;
                }
....

       recordId,
```

there's also a test tour that i wrote for this task, try to include the case you covered here.

Have a nice day

Reply:
Ok i'll try, first time with the test tour :(

Reply:
don't worry, i'm here if you need any help, i will send you the exact file where you need to fix next monday.
Have a nice weekend

Reply:
@RaoufGhrissi Yes please, really appreciate your helpe :pray:

Reply:
hello 
take a look to this PR (the one introduced the feature)
https://github.com/odoo/odoo/commit/8bae9832c0ef0e6bf847ca95e44d25200b1f6c6c

file addons/project/tests/test_project_ui.py is containing the python test, 
in line 26, add watch = True to visualise the tour

```self.start_tour('/web', 'project_task_history_tour', login='admin', watch=True)```

the js tour is in file addons/project/static/tests/tours/project_task_history.js
try to understand the steps and add new ones to cover your case